modules/gdscript/gdscript.cpp

```diff
@@ -50,11 +41,22 @@
 #include "tests/gdscript_test_runner.h"
 #endif
 
+#include "core/config/engine.h"
+#include "core/config/project_settings.h"
+#include "core/core_constants.h"
+#include "core/core_string_names.h"
+#include "core/io/file_access.h"
+#include "core/io/file_access_encrypted.h"
+#include "core/os/os.h"
+
 #ifdef TOOLS_ENABLED
-#include "editor/editor_paths.h"
 #include "editor/gdscript_docgen.h"
+
+#include "editor/editor_paths.h"
 #endif
 
+#include <stdint.h>
```

Comment:
Shouldn't it go before conditional includes?

Reply:
No, the overall logic I go for is:
- Local includes
- If any, conditional local includes
- Core includes
- If any, conditional core includes
- Thirdparty includes
- If any, conditional thirdparty includes

When there are both local and core conditional includes with the same condition (like here), I do:
- Local includes
- Core includes
- Conditional local and core includes in the same condition
- Thirdparty includes

Reply:
And if all 3 groups are conditioned, then I guess it goes to the very bottom? I don't particularly see why we wouldn't put all conditionals after unconditional includes in all cases, but I guess there is some consistency with this approach :)

Let's see how hard it would be to notice and enforce. Another concern would be that if someone needs to put a new core include under the same condition as an existing local include in the first example, we would have to ask them to move the whole section below, or keep an inconsistency.

Reply:
We could keep the logic to the main pattern, and split the gdscript and core includes there, even if they're under the same condition. So like this:
```cpp
#include "gdscript.h"

#include "gdscript_analyzer.h"
#include "gdscript_cache.h"
#include "gdscript_compiler.h"
#include "gdscript_parser.h"
#include "gdscript_rpc_callable.h"
#include "gdscript_warning.h"

#ifdef TOOLS_ENABLED
#include "editor/gdscript_docgen.h"
#endif

#ifdef TESTS_ENABLED
#include "tests/gdscript_test_runner.h"
#endif

#include "core/config/engine.h"
#include "core/config/project_settings.h"
#include "core/core_constants.h"
#include "core/core_string_names.h"
#include "core/io/file_access.h"
#include "core/io/file_access_encrypted.h"
#include "core/os/os.h"

#ifdef TOOLS_ENABLED
#include "editor/editor_paths.h"
#endif

#include <stdint.h>
```

Reply:
I think I like it more, though at this point that should probably be done as a follow-up.

Reply:
Well it's only a handful of cases affected, so I'll amend this PR to do this. It solves the awkward case from https://github.com/godotengine/godot/pull/75944#discussion_r1230708791 to some extent.

Reply:
Pushed it as a separate commit for review: https://github.com/godotengine/godot/pull/75944/commits/0d1c09adc56aa9690dd3cc04bfbd036ae3a4d349